mirdata/billboard.py

```diff
@@ -0,0 +1,453 @@
+        )
+
+    @utils.cached_property
+    def chords(self):
+        return {
+            "full": _load_chords(self.lab_full_path),
+            "majmin7": _load_chords(self.lab_majmin7_path),
+            "majmin7inv": _load_chords(self.lab_majmin7inv_path),
+            "majmin": _load_chords(self.lab_majmin_path),
+            "majmininv": _load_chords(self.lab_majmininv_path),
+        }
+
+    @utils.cached_property
+    def sections(self):
+        return _load_sections(
+            os.path.join(self._data_home, self._track_paths["salami"][0])
+        )
+
+    @property
+    def audio(self):
+        return librosa.load(self.audio_path, sr=None, mono=True)
+
+    def to_jams(self):
+        return jams_utils.jams_converter(
+            chord_data=[
+                (self.chords["full"], "full"),
+                (self.chords["majmin7"], "majmin7"),
+                (self.chords["majmin7inv"], "majmin7inv"),
+                (self.chords["majmin"], "majmin"),
+                (self.chords["majmininv"], "majmininv"),
+            ],
+            section_data=[(self.sections, None)],
+            metadata=self._track_metadata,
+        )
+
+
+def _load_chords(path):
+    """Private function to load LAB format chord data from a file
+
+    Args:
+        chords_path (str):
+
+    """
+
+    if path is None or not os.path.exists(path):
+        return None
+
+    start_times, end_times, chords = [], [], []
+    with open(path, "r") as f:
+        for l in f:
+            l = l.rstrip()
+            if l:
+                start, end, label = l.split("\t")
+                start_times.append(float(start))
+                end_times.append(float(end))
+                chords.append(label)
+
+    chord_data = utils.ChordData(np.array([start_times, end_times]).T, chords)
+    return chord_data
+
+
+def _load_sections(sections_path):
+    """Private function to load SALAMI format sections data from a file
+
+    Args:
+        sections_path (str):
+
+    """
+    if sections_path is None or not os.path.exists(sections_path):
+        return None
+
+    timed_sections = _timed_sections(_parse_salami(sections_path))
+
+    # Clean sections
+    timed_sections_clean = [ts for ts in timed_sections if ts["section"] is not None]
+
+    start_times, end_times, sections = [], [], []
+    for idx, ts in enumerate(timed_sections_clean):
+        if idx < len(timed_sections_clean) - 1:
+            start_times.append(timed_sections_clean[idx]["time"])
+            end_times.append(timed_sections_clean[idx + 1]["time"])
+            sections.append(timed_sections_clean[idx]["section"])
+        else:
+            start_times.append(timed_sections_clean[idx]["time"])
+            end_times.append(timed_sections[-1]["time"])  # end of song
+            sections.append(timed_sections_clean[idx]["section"])
+
+    section_data = utils.SectionData(np.array([start_times, end_times]).T, sections)
+
+    return section_data
+
+
+def _parse_salami(fn):
```

Comment:
Let's leave this for now, but open an issue once this PR is merged to see if there's a more concise way to do this!